libcnb-cargo/src/lib.rs

```diff
@@ -153,33 +152,26 @@ pub struct BuildpackData<BM> {
 ///
 /// # Errors
 ///
-/// Will return `Err` if the buildpack directory already exists or could not be assembled.
+/// Will return `Err` if the buildpack directory could not be assembled.
 pub fn assemble_buildpack_directory(
     destination_path: impl AsRef<Path>,
     buildpack_descriptor_path: impl AsRef<Path>,
     buildpack_binary_path: impl AsRef<Path>,
 ) -> std::io::Result<()> {
-    if destination_path.as_ref().exists() {
-        Err(io::Error::new(
-            io::ErrorKind::AlreadyExists,
-            "Destination path already exists!",
-        ))
-    } else {
-        fs::create_dir_all(destination_path.as_ref())?;
+    fs::create_dir_all(destination_path.as_ref())?;
 
-        fs::copy(
-            buildpack_descriptor_path.as_ref(),
-            destination_path.as_ref().join("buildpack.toml"),
-        )?;
+    fs::copy(
+        buildpack_descriptor_path.as_ref(),
+        destination_path.as_ref().join("buildpack.toml"),
+    )?;
 
-        let bin_path = destination_path.as_ref().join("bin");
-        fs::create_dir_all(&bin_path)?;
+    let bin_path = destination_path.as_ref().join("bin");
+    fs::create_dir_all(&bin_path)?;
 
-        fs::copy(buildpack_binary_path.as_ref(), bin_path.join("build"))?;
-        create_file_symlink("build", bin_path.join("detect"))?;
+    fs::copy(buildpack_binary_path.as_ref(), bin_path.join("build"))?;
+    create_file_symlink("build", bin_path.join("detect"))?;
```

Comment:
Will symlink creation fail if the symlink already exists? If so, does this affect the #277 use-case?

Looking at:
https://doc.rust-lang.org/src/std/os/unix/fs.rs.html#899-901

...it calls:
https://doc.rust-lang.org/src/std/sys/unix/fs.rs.html#1136

And these libc docs seem to suggest that could fail if the file already exists:
https://ftp.gnu.org/old-gnu/Manuals/glibc-2.2.3/html_node/libc_269.html

> EEXIST
>    There is already an existing file named newname.

Reply:
Sure, but a similar concern would also apply to all other FS/IO operations in that function. I think it's perfectly fine to not add additional error handling for these cases. `assemble_buildpack_directory` already returns `std::io::Result` which can signal any IO (and FS) error that's happening. It's not like the callsite can reasonably recover anyway.

#277 will always pass an empty directory, it will not run into the specific issue you're mentioning.

Reply:
I wasn't meaning additional error handling - it just wasn't clear whether the directory would be empty or not - though sounds like it will be, so this doesn't apply.